src/elements/Pagination/index.jsx

```diff
@@ -0,0 +1,199 @@
+    const res = []
+
+    if ((currentPage - 1) > leftLength) {
+      for (let i = 0; i < leftLength; i += 1) {
+        if (i === leftLength - 1) {
+          res.unshift(1)
+        } else if (i === leftLength - 2) {
+          res.unshift('ellipsis')
+        } else {
+          res.unshift(currentPage - i - 1)
+        }
+      }
+    } else {
+      for (let i = 0; i < leftLength; i += 1) {
+        res.push(i + 1)
+      }
+    }
+    return res;
+  }
+
+  calculateFollowingSteps (rightLength, currentPage, totalPages) {
+    const res = []
+
+    if ((totalPages - currentPage) > rightLength) {
+      for (let i = 0; i < rightLength; i += 1) {
+        if (i === rightLength - 1) {
+          res.push(totalPages)
+        } else if (i === rightLength - 2) {
+          res.push('ellipsis')
+        } else {
+          res.push(currentPage + i + 1)
+        }
+      }
+    } else {
+      for (let i = 0; i < rightLength; i += 1) {
+        res.push(currentPage + i + 1)
+      }
+    }
+    return res;
+  }
+
+  renderPagination () {
+    const { currentPage, totalPages } = this.state
+    if (currentPage < 1) {
+      this.changePage(1)
+    } else if (currentPage > totalPages) {
+      this.changePage(totalPages)
+    }
+
+    const paginations = []
+    const steps = this.calculateSteps();
+
+    for (let i = 0; i < steps.length; i += 1) {
+      const label = steps[i];
+      let isClickable = true;
+
+      if (currentPage === steps[i]) {
+        isClickable = false;
+      }
+      if (label === 'ellipsis') {
+        paginations.push(<PaginationLink isPageBreak={true} key={i} />)
+      } else {
+        paginations.push(
+          <PaginationLink
+            label={label}
+            isClickable={isClickable}
+            onClick={(page) => { this.onClick(page) }}
+            key={i} />
+        )
+      }
+    }
+
+    return paginations;
+  }
+
+  renderLeftCaret () {
+    return (
+      <li className="Pagination__caret">
+        { this.state.currentPage <= 1
+          ? <span>{'<'}</span>
+          : <a onClick={() => {
+            this.onClick(this.state.currentPage - 1)
+          }}>
+            {'<'}
+          </a> }
+      </li>
+    )
+  }
+
+  renderRightCaret () {
+    return (
+      <li className="Pagination__caret">
+        { this.state.currentPage >= this.state.totalPages
+          ? <span>{'>'}</span>
+          : <a onClick={() => {
+            this.onClick(this.state.currentPage + 1)
+          }}>
+            {'>'}
+          </a>
+        }
+      </li>
+    )
+  }
+
+  render () {
+    const leftCaret = this.renderLeftCaret();
+    const rightCaret = this.renderRightCaret();
+    const pagination = this.renderPagination();
+
+    return (
+      <div className={classNames(
+        'Pagination',
+        `Pagination--${this.props.fontSize}`)}>
+        <ul className={classNames('Pagination__list')}>
+          {leftCaret}
+          {pagination}
+          {rightCaret}
```

Comment:
can you think of using `icon` for caret to provide more option for users?

Reply:
Do you have idea which icon I can use for it?

Reply:
<img width="985" alt="screen shot 2017-09-18 at 8 20 53 pm" src="https://user-images.githubusercontent.com/1316039/30574480-03a57d90-9caf-11e7-9778-e63b94b57a79.png">

I just tried to make some example based on your branch. What I mean is this is general toolset, not for single usage. So, I think give option for flexibility is good for users. I don't mean you need to use specific icon.

Reply:
I see what you mean. That's good to let user use their custom style. I will add the interface.